### PR TITLE
Resolve issue with simultaneous recognition of tap gesture

### PR DESCRIPTION
**Pull request by ZiZasaurus**
Resolves an issue with the `singleTapGestureHandler` not recognizing taps on the mapview ([1697](https://github.com/mapbox/mapbox-maps-ios/issues/1697)) after implementing isSelected/isDraggable [feature](https://github.com/mapbox/mapbox-maps-ios/pull/1659).  A gesture recognizer method has been added to the `singleTapGestureHandler` to allow simultaneous recognition when using the tap gesture.

## Pull request checklist:
 - [x] Describe the changes in this PR, especially public API changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [x] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` first and then port to `v10.[version]` release branch.

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Review by evil159 — CHANGES_REQUESTED** [review]
The fix looks good 👍 Tests are missing though 😕

File: Sources/MapboxMaps/Gestures/GestureHandlers/SingleTapGestureHandler.swift [hunk 1]
```diff
@@ -25,3 +26,9 @@ internal final class SingleTapGestureHandler: GestureHandler {
         }
     }
 }
+
+extension SingleTapGestureHandler: UIGestureRecognizerDelegate {
+    func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        return gestureRecognizer is UITapGestureRecognizer
```
Comment (maios) [inline Sources/MapboxMaps/Gestures/GestureHandlers/SingleTapGestureHandler.swift:32]: should we also check condition of `otherGestureRecognizer` being of type `UITapGestureRecognizer`?
and maybe a check `gestureRecognizer === self.gestureRecognizer` is better here and more consistent to other GestureHandler i.e https://github.com/mapbox/mapbox-maps-ios/blob/main/Sources/MapboxMaps/Gestures/GestureHandlers/PinchGestureHandler.swift#L124

**Comment by pjleonard37** [comment]
👍  - this resolved the issue for me. Do we need to adjust the other gesture handlers as well? I'm specifically thinking about LongPress, but others may need to be adjusted. 

Additionally, can we expand tests across the gesture handlers to catch these behavior changes?

**Review by persidskiy — APPROVED** [review]
lgtm

**Review by maios — APPROVED** [review]
🎉

**Review by evil159 — APPROVED** [review]
🔥

**Review by pjleonard37 — APPROVED** [review]

**Pull request merged**